audit/parse.go

```diff
@@ -0,0 +1,88 @@
+package audit
+
+import (
+	"crypto"
+	"encoding/base64"
+	"encoding/pem"
+	"errors"
+	"fmt"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/google/certificate-transparency-go/tls"
+	"github.com/google/certificate-transparency-go/x509"
+)
+
+// ParseChainFromPEM parses an X509 certificate from data in PEM format.
+func ParseCertificateFromPEM(data []byte) (*x509.Certificate, error) {
+	block, rest := pem.Decode(data)
+	if len(rest) != 0 {
+		return nil, errors.New("trailing data found after PEM block")
+	}
+	if block == nil {
+		return nil, errors.New("PEM block is nil")
+	}
+	if block.Type != "CERTIFICATE" {
+		return nil, errors.New("PEM block is not a CERTIFICATE")
+	}
+
+	cert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return cert, nil
+}
+
+// ParseChainFromPEM parses an X509 certificate chain from data in PEM format.
+func ParseChainFromPEM(data []byte) ([]*x509.Certificate, error) {
+	var chain []*x509.Certificate
+	for {
```

Comment:
similar code in `client/addChain`, refactor?

Reply:
Turns out you mean ctclient/addchain.  Yup this could be refactored to use what David suggested, but doesn't quite fit to use what this PR adds, so leaving for a future PR.